docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_payment_failed_remediation.mdx

```diff
@@ -1,6 +1,229 @@
+           else:
+               print(f"Reserved Node ID {node['ReservedNodeId']} is in good standing.")
+   ```
+
+By following these steps, you can proactively monitor and prevent Redshift Reserved Nodes from having a "Payment Failed" status using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard. From the navigation pane, choose "Reserved Nodes".
+3. In the "Reserved Nodes" section, you will see a list of all your reserved nodes. Check the "Status" column of each reserved node.
+4. If the status of any reserved node is "payment-failed", it means that the payment for that reserved node has failed.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift Reserved Nodes: Use the following AWS CLI command to list all your Redshift Reserved Nodes:
+
+   ```
+   aws redshift describe-reserved-nodes
+   ```
+
+   This command will return a JSON output with details about all your Redshift Reserved Nodes.
+
+3. Parse the output: You need to parse the JSON output to find the nodes with the status "payment-failed". You can do this using a tool like `jq` or a scripting language like Python. Here is an example using `jq`:
+
+   ```
+   aws redshift describe-reserved-nodes | jq -r '.ReservedNodes[] | select(.State=="payment-failed")'
+   ```
+
+4. Check the status: If the above command returns any output, it means that there are Redshift Reserved Nodes with the status "payment-failed". If it doesn't return any output, it means that all your Redshift Reserved Nodes have a status other than "payment-failed".
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): To interact with AWS services, you need to install and configure Boto3. You can install it using pip:
+
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to enable Boto3 to communicate with AWS services:
+
+   ```
+   aws configure
+   ```
+   You'll be prompted to provide your AWS Access Key ID, Secret Access Key, default region name, and default output format.
+
+2. Import necessary libraries and establish a session: Import the Boto3 library into your Python script and establish a session using your credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>